src/com/phonebookservice/database/ContactsFileParser.java

```diff
@@ -0,0 +1,82 @@
+package com.phonebookservice.database;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+
+import com.phonebookservice.exception.InternalServerException;
+import com.phonebookservice.model.Contact;
+import com.phonebookservice.server.ContactConverter;
+import com.phonebookservice.util.CollectionUtility;
+
+public class ContactsFileParser implements IContactsParser {
+    private static final String ENCODING = "UTF-8";
+    private final String fileName;
+
+    /**
+     * Initialization of file parser.
+     *
+     * @param fileName the file name.
+     */
+    public ContactsFileParser(final String fileName) {
+        this.fileName = fileName;
+    }
+
+    @Override
+    public final Collection<Contact> readContacts() {
+        final List<Contact> contacts = new ArrayList<>();
+
+        try {
+            final File file = new File(this.fileName);
+            final List<String> lines = FileUtils.readLines(file,
+                    ContactsFileParser.ENCODING);
+
+            for (String line : lines) {
+                final String[] splitLine = line
+                        .split(ContactConverter.SPLITTER);
+                final Contact contact = ContactConverter
+                        .convertStringToContact(splitLine);
+                contacts.add(contact);
+            }
+        } catch (IOException e) {
+            throw new InternalServerException(e);
```

Comment:
Why internalServer??
Is a converter is a service??
Database is a service 
ControlPlan is a service

Converter not a service to throw serviceExceotion

Reply:
you are right but we should throw a generic internal exception